### ISSUE TITLE
System.Diagnostics.ActivitySource timestamps are not accurate on .NET Framework

### ISSUE THREAD

**Issue by ghelyar** [dotnet/runtime#44354]
### Description

[Despite using a Stopwatch](https://github.com/dotnet/runtime/blob/3ac170acc89adde016b1297f7c29232cda61c102/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.DateTime.netfx.cs), Activity start times and durations are not accurate. It's possible to run several activities one after the other, and get overlapping start and stop times. Even an empty activity can report ~9ms duration.

### Configuration

.net 4.7.2, Windows 10, x64. Also seen on Windows Server 2019.

### Other information

It's a bit of a hard one to get a proper repro, but it's easy to test with P/Invoke to `GetSystemTimePreciseAsFileTime` and setting a tag on the Activity for comparison to the start/end times that are recorded - sometimes the time is not even inside the start-end range that are reported. With the following code, a sawtooth can be easily seen in the results:

```C#
using (var listener = new ActivityListener
{
	ShouldListenTo = _ => true,
	Sample = (ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.AllDataAndRecorded
})
{
	ActivitySource.AddActivityListener(listener);
	var source = new ActivitySource("test");
	for (var i = 0; i <= 1000; ++i)
	{
		Activity a;
		using (var activity = source.StartActivity("test.activity"))
		{
			a = activity;
		}
		if (i > 0)
		{
			Console.WriteLine(a.Duration.Ticks);
		}
	}
}
```

![image](https://user-images.githubusercontent.com/3225358/98378619-2f62d700-203e-11eb-9629-bd6375eb4097.png)

When I display these spans on a graph, the overlap causes them to be displayed strangely, rather than as nice flames/icicles:
![image](https://user-images.githubusercontent.com/3225358/98378827-70f38200-203e-11eb-9f47-18f1ec10724c.png)

My current workaround is to wrap activities, so that the start and end times are updated just after creation and just before disposal, and this works for things I have manually instrumented myself, but not for automatic instrumentation like `OpenTelemetry.Instrumentation.Http`:

```C#
public sealed class HighPrecisionActivity : IDisposable
{
	public HighPrecisionActivity(Activity activity)
	{
		Activity = activity ?? throw new ArgumentNullException(nameof(activity));
		activity.SetStartTime(UtcNow);
	}

	public Activity Activity { get; }

	public void Dispose()
	{
		Activity.SetEndTime(UtcNow);
		Activity.Dispose();
	}
	
	private static DateTime UtcNow
	{
		get
		{
			GetSystemTimePreciseAsFileTime(out long filetime);
			return DateTime.FromFileTimeUtc(filetime);
		}
	}
	
	[DllImport("Kernel32.dll", CallingConvention = CallingConvention.Winapi)]
	private static extern void GetSystemTimePreciseAsFileTime(out long filetime);
}
```

if I do this, and then compare the start/end times vs the start/end times reported by `ActivitySource` itself, by using AddTag instead of `SetStartTime` and `SetEndTime` then you can see how far out it is:

![image (1)](https://user-images.githubusercontent.com/3225358/98379973-db58f200-203f-11eb-8c94-256bdf53dff6.png)
the horizontal axis is time

**Comment by tarekgh** [comment]
https://github.com/dotnet/runtime/issues/45111

**Comment by tarekgh** [comment]
@ghelyar I am seeing you are using .NET Framework. I have a couple of questions I hope you can help with:
- If you use `new Activity(...)` instead of `ActivitySource.StartActivity(...)`, do you still see the problem in your environment? I am not suggesting you do that in your real code but trying to get more info.
- If you use .NET Core, do you see the same problem?

Thanks!

**Comment by tarekgh** [comment]
This is fixed by the PR #45176 and the fix will be released in the next servicing release 5.0.1.
